whois-api/src/main/java/net/ripe/db/whois/api/UpdatesParser.java

```diff
@@ -62,12 +63,15 @@ public List<Paragraph> createParagraphs(final ContentWithCredentials contentWith
             baseCredentials.add(SsoCredential.createOfferedCredential(updateContext.getUserSession()));
         }
 
-        updateContext.getClientCertificate().ifPresent(x509 -> baseCredentials.add(ClientCertificateCredential.createOfferedCredential(x509)));
+        if (updateContext.getClientCertificates() != null) {
```

Comment:
Just a suggestion
I think that this code is duplicated in some other classes. Maybe we can put this code just in a common class and call it from the rest.

Same happen with this method:
 public void setClientCertificates(final UpdateContext updateContext, final HttpServletRequest request) {
    updateContext.setClientCertificates(ClientCertificateExtractor.getClientCertificates(request));
}

However, it is not too much duplicated code so feel free to ignore

Reply:
If you look at the 3 places this code is duplicated, there is not much overlap (it's used for different types of updates) so de-duplicating those lines are not straightforward. It's also the case for the other methods.